src/fields.jl

```diff
@@ -14,13 +14,21 @@ A field datatype for 1D objects.
 
 $(TYPEDFIELDS)
 """
-struct Field1D{LX<:AbstractLocation} <: AbstractField
+struct Field1D{LX<:AbstractLocation, G} <: AbstractField
     "Array with the values of the field."
-    data :: Array
+    data::AbstractArray
     "The grid on which the field lives."
-    grid :: Grid1D
+    grid :: G
+    
+    #Field1D(LX, data, grid) = new{LX, G}(data, grid)    
```

Comment:
Commented out code is bad practice. If this is not needed then delete?

Reply:
I wanted to keep it till we merge this PR, in case we prefer to have an inner constructor.

Reply:
OK, let's see... Does it hurt if we keep the inner constructor also?

I guess we can keep the inner constructor but the users will have to provide the `data` as an `OffsetArray`, which is cumbersome. Do we want that? If not, then I can't think of any other reason we might want an inner constructor...

Now playing devil's advocate, _with the inner constructor in place_, the outer constructor could be slightly simplified. Specifically the line `field = Field1D{LX, typeof(grid)}(data_with_halos, grid)` could be replaced with `field = Field1D(LX, data_with_halos, grid)`. That's nice, no?